lib/Jsrt/JsrtDebugManager.cpp

```diff
@@ -364,8 +364,21 @@ void JsrtDebugManager::CallDebugEventCallback(JsDiagDebugEvent debugEvent, Js::D
 
     auto funcPtr = [&]()
     {
-        AutoClear autoClear(this);
-        this->debugEventCallback(debugEvent, eventDataObject, this->callbackState);
+        if (isBreak)
+        {
+            // If we are reporting break we should clear all objects after call returns
+            AutoClear autoClear(this);
+
+            // Save the frame address, when asking for stack we will only give stack which is under this address
+            // because host can execute javascript after break which should not be part of stack.
+            void *frameAddress = _AddressOfReturnAddress();
+            scriptContext->GetThreadContext()->GetDebugManager()->SetDispatchHaltFrameAddress(frameAddress);
```

Comment:
It would be good idea to do SetDispatchHaltFrameAddress(nullptr) after the debugEventCallback call below, so that threadContext->GetDebugManager()->GetDispatchHaltFrameAddress() can return null once the break is over.